hoomd/mpcd/CollisionMethod.cc

```diff
@@ -48,15 +78,945 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
     // sync the embedded group
     m_cl->setEmbeddedGroup(m_embed_group);
 
+    // check for collision warnings the first time collide is run
+    if (!m_checked_collision_warnings)
+        {
+        checkCollisionWarnings(timestep);
+        m_checked_collision_warnings = true;
+        }
+    if (m_needs_temperature && !m_T)
+        {
+        throw std::runtime_error("Temperature required by collision method");
+        }
+
+    // check the GPU autotuners
+#ifdef ENABLE_HIP
+    if (m_check_rigid_tuners)
+        {
+        checkRigidAutotuners();
+        m_check_rigid_tuners = false;
+        }
+#endif // ENABLE_HIP
+
     // set random grid shift
     m_cl->drawGridShift(timestep);
 
     // update cell list
     m_cl->compute(timestep);
 
+    // create auxillary array for rigid bodies
```

Comment:
This comment is not quite descriptive of what is happening here:
```suggestion
    // setup rigid bodies before collision happens 
```